File: TicketCatalogueService/src/main/kotlin/it/polito/wa2/lab5/group09/ticketcatalogueservice/entities/Order.kt
package it.polito.wa2.lab5.group09.ticketcatalogueservice.entities

import it.polito.wa2.lab5.group09.ticketcatalogueservice.controllers.TravelcardOwnerDTO
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document



@Document(collection = "orders")
data class Order(
    @Id
    @Indexed
    val orderId: ObjectId = ObjectId.get(),
    var status: Status = Status.PENDING,
    val ticketCatalogueId: ObjectId,
    val quantity: Int,
    val customerUsername: String,
    val travelcardOwner: TravelcardOwnerDTO? = null
)

enum class Status {
    PENDING, CANCELED, ACCEPTED
}

File: client/courseprojectg09/src/model/Order.js
class Order {
    constructor(orderId, status, ticketCatalogueId, duration, quantity, customerUsername, owner) {
        this.orderId = orderId;
        this.status = status;
        this.ticketCatalogueId = ticketCatalogueId;
        this.duration = duration;
        this.quantity = quantity;
        this.customerUsername = customerUsername;
        this.owner = owner;
    }

    static from(json) {
        const order = new Order();
        Object.assign(order, json);
        return order;
    }
}

export default Order;